pkg/pmem-csi-driver/nodeserver.go

```diff
@@ -129,7 +124,7 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 		req.GetVolumeId(), targetPath, srcPath, readOnly, mountFlags, fsType)
 
 	// Kubernetes v1.16+ would request ephemeral volumes via VolumeContext
-	val, ok := req.GetVolumeContext()[volumeContextEphemeral]
+	val, ok := req.GetVolumeContext()[parameterEphemeral]
```

Comment:
`"csi.storage.k8s.io/ephemeral"`is not a parameter passed by the end-user. Better not to mix with parameter handling.

Reply:
"parameter" does not mean that is has to come directly from an end-user. I was using it in the more general meaning of "attributes" of a volume. With that in mind, the name `parameterEphemeral` makes sense to me. :shrug:

Regarding "mixing": we don't have a choice here. This `req.GetVolumeContext()` can contain parameters set by us (in `CreateVolume` and thus indirectly by the user), parameters set by a user (the inline volume parameters) and by Kubernetes (pod info, parameterEphemeral).

I think it makes sense to use the same `parameter` prefix for all of those strings to document that they all appear as keys in the same map.

Reply:
@pohly Ok, I got your idea of treating volume contexts also like parameters. This case better to parse/validate the volume context bit beforehand so that we can avoid special treatment for `parameterEphemeral` and `volumeProvisionerIdentity/parameterProvisionerID`.

I would propose to change `ephemeralVolumeParameters` origin to something like `publishVolumeParameters` and parse them beforehand and use `params.Ephemeral()` params.ProvisionerID()` calls

Reply:
But parameters for publishing of ephemeral volumes and for persistent volumes are not the same. With your proposal, `publishVolumeParameters` would have to be a superset of both and then we won't catch when a user erroneously puts "cacheSize" into the inline volume spec. That's why we need to validate differently after determining what the volume is.

Reply:
> we won't catch when a user erroneously puts "cacheSize" into the inline volume spec.

Yes, agree.